jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/delegate/AbstractGridDelegate.java

```diff
@@ -197,6 +199,28 @@ && getComponent().getEditor().isOpen()) {
         }
     }
 
+    /**
+     * Refreshes current selection using provided items.
+     */
+    protected void refreshSelection(Collection<E> items) {
+        Set<E> prevSelectedItemsToRefresh = new HashSet<>(getSelectedItems());
+
+        List<E> itemsToSelect = new ArrayList<>(prevSelectedItemsToRefresh.size());
+        for (E item : items) {
+            //select the item if it was selected before refresh
+            if (prevSelectedItemsToRefresh.remove(item)) {
+                itemsToSelect.add(item);
+            }
+
+            //skip further checks if no more items were selected
+            if (prevSelectedItemsToRefresh.isEmpty()) {
+                break;
+            }
+        }
+        deselectAll();
```

Comment:
let's add a detailed description why we do this, because it will cause in several selection events that may cause problems if users use selection event to load related data